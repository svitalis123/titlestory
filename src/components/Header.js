import React from 'react';
import { HiChevronDoubleRight } from 'react-icons/hi';
import { BsBook } from 'react-icons/bs';
import { VscUnlock } from 'react-icons/vsc';
import { MdKeyboardArrowDown } from 'react-icons/md';
import styles from '../css/Header.module.css';

// Header function exported as default renders header content
export default function Header() {
  return (
    <header className={styles.header}>
      <div className={styles.headerdiv1}>
        <HiChevronDoubleRight className={styles.headerfirsticon} />
        <nav className={styles.headerdiv1nav}>
          <BsBook className={styles.headersecondiconbook} />
          <button type="button" className={styles.headernavbutton1}>Main</button>
          <span className={styles.headernavspans}>/</span>
          <button type="button" className={styles.headernavbuttons}>Getting started</button>
          <span className={styles.headernavspans}>/</span>
          <button type="button" className={styles.headernavbuttons}>Front-end developer test proje...</button>
        </nav>
      </div>
      <div className={styles.headerdiv2}>
        <VscUnlock className={styles.headerbutton2diviconlock} />
        <button type="button" className={styles.headernavbuttons}>Editing</button>
        <span className={styles.headernavspans}>|</span>
        <button type="button" className={styles.headerseconddivpublishbtn}>
          Publish Space
          <MdKeyboardArrowDown className={styles.headerbuton2div2caretdown} />
        </button>
      </div>
    </header>
  );
}
