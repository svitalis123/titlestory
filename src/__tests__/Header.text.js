import React from 'react';
import { render } from '@testing-library/react';
import Header from '../components/Header';

describe('test the rendering of header function', () => {
  test('Render header elements on page', () => {
    const display = render(
      <React.StrictMode>
        <Header />
      </React.StrictMode>,
    );
    expect(display).toMatchSnapshot();
  });
});
