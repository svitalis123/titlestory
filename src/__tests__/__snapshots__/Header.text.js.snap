// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test the rendering of header function Render header elements on page 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header
        class="header"
      >
        <div
          class="headerdiv1"
        >
          <svg
            class="headerfirsticon"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 20 20"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M10.293 15.707a1 1 0 010-1.414L14.586 10l-4.293-4.293a1 1 0 111.414-1.414l5 5a1 1 0 010 1.414l-5 5a1 1 0 01-1.414 0z"
              fill-rule="evenodd"
            />
            <path
              clip-rule="evenodd"
              d="M4.293 15.707a1 1 0 010-1.414L8.586 10 4.293 5.707a1 1 0 011.414-1.414l5 5a1 1 0 010 1.414l-5 5a1 1 0 01-1.414 0z"
              fill-rule="evenodd"
            />
          </svg>
          <nav
            class="headerdiv1nav"
          >
            <svg
              class="headersecondiconbook"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 16 16"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M1 2.828c.885-.37 2.154-.769 3.388-.893 1.33-.134 2.458.063 3.112.752v9.746c-.935-.53-2.12-.603-3.213-.493-1.18.12-2.37.461-3.287.811V2.828zm7.5-.141c.654-.689 1.782-.886 3.112-.752 1.234.124 2.503.523 3.388.893v9.923c-.918-.35-2.107-.692-3.287-.81-1.094-.111-2.278-.039-3.213.492V2.687zM8 1.783C7.015.936 5.587.81 4.287.94c-1.514.153-3.042.672-3.994 1.105A.5.5 0 0 0 0 2.5v11a.5.5 0 0 0 .707.455c.882-.4 2.303-.881 3.68-1.02 1.409-.142 2.59.087 3.223.877a.5.5 0 0 0 .78 0c.633-.79 1.814-1.019 3.222-.877 1.378.139 2.8.62 3.681 1.02A.5.5 0 0 0 16 13.5v-11a.5.5 0 0 0-.293-.455c-.952-.433-2.48-.952-3.994-1.105C10.413.809 8.985.936 8 1.783z"
              />
            </svg>
            <button
              class="headernavbutton1"
              type="button"
            >
              Main
            </button>
            <span
              class="headernavspans"
            >
              /
            </span>
            <button
              class="headernavbuttons"
              type="button"
            >
              Getting started
            </button>
            <span
              class="headernavspans"
            >
              /
            </span>
            <button
              class="headernavbuttons"
              type="button"
            >
              Front-end developer test proje...
            </button>
          </nav>
        </div>
        <div
          class="headerdiv2"
        >
          <svg
            class="headerbutton2diviconlock"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 16 16"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M5 7V5a3 3 0 0 1 5.83-1h1.044A4.002 4.002 0 0 0 4 5v2H3L2 8v6l1 1h10l1-1V8l-1-1H5zm6 1h2v6H3V8h8z"
              fill-rule="evenodd"
            />
          </svg>
          <button
            class="headernavbuttons"
            type="button"
          >
            Editing
          </button>
          <span
            class="headernavspans"
          >
            |
          </span>
          <button
            class="headerseconddivpublishbtn"
            type="button"
          >
            Publish Space
            <svg
              class="headerbuton2div2caretdown"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 0h24v24H0V0z"
                fill="none"
              />
              <path
                d="M7.41 8.59L12 13.17l4.59-4.58L18 10l-6 6-6-6 1.41-1.41z"
              />
            </svg>
          </button>
        </div>
      </header>
    </div>
  </body>,
  "container": <div>
    <header
      class="header"
    >
      <div
        class="headerdiv1"
      >
        <svg
          class="headerfirsticon"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 20 20"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M10.293 15.707a1 1 0 010-1.414L14.586 10l-4.293-4.293a1 1 0 111.414-1.414l5 5a1 1 0 010 1.414l-5 5a1 1 0 01-1.414 0z"
            fill-rule="evenodd"
          />
          <path
            clip-rule="evenodd"
            d="M4.293 15.707a1 1 0 010-1.414L8.586 10 4.293 5.707a1 1 0 011.414-1.414l5 5a1 1 0 010 1.414l-5 5a1 1 0 01-1.414 0z"
            fill-rule="evenodd"
          />
        </svg>
        <nav
          class="headerdiv1nav"
        >
          <svg
            class="headersecondiconbook"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 16 16"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M1 2.828c.885-.37 2.154-.769 3.388-.893 1.33-.134 2.458.063 3.112.752v9.746c-.935-.53-2.12-.603-3.213-.493-1.18.12-2.37.461-3.287.811V2.828zm7.5-.141c.654-.689 1.782-.886 3.112-.752 1.234.124 2.503.523 3.388.893v9.923c-.918-.35-2.107-.692-3.287-.81-1.094-.111-2.278-.039-3.213.492V2.687zM8 1.783C7.015.936 5.587.81 4.287.94c-1.514.153-3.042.672-3.994 1.105A.5.5 0 0 0 0 2.5v11a.5.5 0 0 0 .707.455c.882-.4 2.303-.881 3.68-1.02 1.409-.142 2.59.087 3.223.877a.5.5 0 0 0 .78 0c.633-.79 1.814-1.019 3.222-.877 1.378.139 2.8.62 3.681 1.02A.5.5 0 0 0 16 13.5v-11a.5.5 0 0 0-.293-.455c-.952-.433-2.48-.952-3.994-1.105C10.413.809 8.985.936 8 1.783z"
            />
          </svg>
          <button
            class="headernavbutton1"
            type="button"
          >
            Main
          </button>
          <span
            class="headernavspans"
          >
            /
          </span>
          <button
            class="headernavbuttons"
            type="button"
          >
            Getting started
          </button>
          <span
            class="headernavspans"
          >
            /
          </span>
          <button
            class="headernavbuttons"
            type="button"
          >
            Front-end developer test proje...
          </button>
        </nav>
      </div>
      <div
        class="headerdiv2"
      >
        <svg
          class="headerbutton2diviconlock"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 16 16"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M5 7V5a3 3 0 0 1 5.83-1h1.044A4.002 4.002 0 0 0 4 5v2H3L2 8v6l1 1h10l1-1V8l-1-1H5zm6 1h2v6H3V8h8z"
            fill-rule="evenodd"
          />
        </svg>
        <button
          class="headernavbuttons"
          type="button"
        >
          Editing
        </button>
        <span
          class="headernavspans"
        >
          |
        </span>
        <button
          class="headerseconddivpublishbtn"
          type="button"
        >
          Publish Space
          <svg
            class="headerbuton2div2caretdown"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 0h24v24H0V0z"
              fill="none"
            />
            <path
              d="M7.41 8.59L12 13.17l4.59-4.58L18 10l-6 6-6-6 1.41-1.41z"
            />
          </svg>
        </button>
      </div>
    </header>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
